src/lvl_script_commands_old.c

```diff
@@ -1159,45 +1159,48 @@ static void command_export_variable(long plr_range_id, const char *varib_name, c
 
 static void command_use_spell_on_creature(long plr_range_id, const char *crtr_name, const char *criteria, const char *magname, int splevel)
 {
-  SCRIPTDBG(11, "Starting");
-  long mag_id = get_rid(spell_desc, magname);
-  if (splevel < 1)
-  {
-    if ( (mag_id == SplK_Heal) || (mag_id == SplK_Armour) || (mag_id == SplK_Speed) || (mag_id == SplK_Disease) || (mag_id == SplK_Invisibility) || (mag_id == SplK_Chicken) )
+    SCRIPTDBG(11, "Starting");
+    long mag_id = get_rid(spell_desc, magname);
+    if (mag_id == -1)
     {
-        SCRPTWRNLOG("Spell %s level too low: %d, setting to 1.", magname, splevel);
+        SCRPTERRLOG("Unknown magic, '%s'", magname);
+        return;
     }
-    splevel = 1;
-  }
-  if (splevel > (MAGIC_OVERCHARGE_LEVELS+1)) //Creatures cast spells from level 1 to 10, but 10=9.
-  {
-    SCRPTWRNLOG("Spell %s level too high: %d, setting to %d.", magname, splevel, (MAGIC_OVERCHARGE_LEVELS+1));
-    splevel = MAGIC_OVERCHARGE_LEVELS;
-  }
-  splevel--;
-  if (mag_id == -1)
-  {
-    SCRPTERRLOG("Unknown magic, '%s'", magname);
-    return;
-  }
-  long crtr_id = parse_creature_name(crtr_name);
-  if (crtr_id == CREATURE_NONE) {
-    SCRPTERRLOG("Unknown creature, '%s'", crtr_name);
-    return;
-  }
-  long select_id = parse_criteria(criteria);
-  if (select_id == -1) {
-    SCRPTERRLOG("Unknown select criteria, '%s'", criteria);
-    return;
-  }
-  // SpellKind sp = mag_id;
-  // encode params: free, magic, caster, level -> into 4xbyte: FMCL
-  long fmcl_bytes;
-  {
-      signed char m = mag_id, lvl = splevel;
-      fmcl_bytes = (m << 8) | lvl;
-  }
-  command_add_value(Cmd_USE_SPELL_ON_CREATURE, plr_range_id, crtr_id, select_id, fmcl_bytes);
+    long crtr_id = parse_creature_name(crtr_name);
+    if (crtr_id == CREATURE_NONE)
+    {
+        SCRPTERRLOG("Unknown creature, '%s'", crtr_name);
+        return;
+    }
+    long select_id = parse_criteria(criteria);
+    if (select_id == -1)
+    {
+        SCRPTERRLOG("Unknown select criteria, '%s'", criteria);
+        return;
+    }
+    struct SpellConfig *spconf = get_spell_config(mag_id);
+    if (spconf->linked_power) // Only check for spells linked to a keeper power.
+    {
+        if (splevel < 1)
+        {
+            SCRPTWRNLOG("Spell %s level too low: %d, setting to 1.", magname, splevel);
+            splevel = 1;
+        }
+        if (splevel > (MAGIC_OVERCHARGE_LEVELS + 1)) // Creatures cast spells from level 1 to 10.
+        {
+            SCRPTWRNLOG("Spell %s level too high: %d, setting to %d.", magname, splevel, (MAGIC_OVERCHARGE_LEVELS + 1));
+            splevel = MAGIC_OVERCHARGE_LEVELS;
+        }
+    }
+    splevel--; // Why is it required? Should it applies only on spells linked to a keeper power?
```

Comment:
This is because in code things are counted from 0 instead of from 1.
The weird fmcl_bytes are no longer needed either if that is your question, but the script command needs to be refactored for it to work with separate variables. This was a hack from when we could not.